File: course_101/exercises/easy_9/double_doubles.rb
# A double number is a number with an even number of digits whose left-side digits are exactly the same as its right-side digits.
# For example, 44, 3333, 103103, 7676 are all double numbers. 444, 334433, and 107 are not.

# Write a method that returns 2 times the number provided as an argument,
# unless the argument is a double number; double numbers should be returned as-is.

def double_number?(num)
  digits = num.digits
  if digits.length.even?
    first_half = digits[0, digits.length / 2]
    second_half = digits[(digits.length / 2)..-1]

    first_half == second_half
  else false
  end
end

def twice(num)
  if double_number?(num) then num
  else num * 2
  end
end

puts twice(2) == 4
puts twice(37) == 74
puts twice(44) == 44
puts twice(334433) == 668866
puts twice(444) == 888
puts twice(107) == 214
puts twice(103103) == 103103
puts twice(3333) == 3333
puts twice(7676) == 7676
puts twice(123_456_789_123_456_789) == 123_456_789_123_456_789
puts twice(5) == 10

File: course_js210_computational_thinking_and_javascript_programming/exercises/js210_small_problems/easy_2/double_doubles.js
// A double number is an even-length number whose left-side digits are exactly the same as its right-side digits. For example, 44, 3333, 103103, and 7676 are all double numbers, whereas 444, 334433, and 107 are not.

// Write a function that returns the number provided as an argument multiplied by two, unless the argument is a double number; return double numbers as-is.

function isDoubleNumber(number) {
  let digits = String(number);
  let center = digits.length / 2;

  leftDigits = digits.slice(0, center);
  rightDigits = digits.slice(center, digits.length);

  console.log(leftDigits)
  console.log(rightDigits)

  return leftDigits === rightDigits;
}

function twice(number) {
  if (isDoubleNumber(number)) {
    return number;
  } else {
    return number * 2;
  }
}


console.log(twice(37));          // 74
console.log(twice(44));          // 44
console.log(twice(334433));      // 668866
console.log(twice(444));         // 888
console.log(twice(107));         // 214
console.log(twice(103103));      // 103103
console.log(twice(3333));        // 3333
console.log(twice(7676));        // 7676